core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -295,10 +297,10 @@ public void registerSuggester(String key, Suggester<?> suggester) {
      * the same name as the builder returns for getWriteableName.</li>
      * </ul>
      */
-    public <T extends ScoreFunctionBuilder<T>> void registerScoreFunction(ParseField xContentField, ScoreFunctionParser<T> parser,
+    public <T extends ScoreFunctionBuilder<T>> void registerScoreFunction(ParseField parserName, ScoreFunctionParser<T> parser,
             String writeableName, Writeable.Reader<T> reader) {
-        for (String name: xContentField.getAllNamesIncludedDeprecated()) {
-            Object oldValue = scoreFunctionParsers.putIfAbsent(name, parser);
+        for (String name: parserName.getAllNamesIncludedDeprecated()) {
+            Object oldValue = scoreFunctionParsers.putIfAbsent(name, new Tuple<>(parserName, parser));
```

Comment:
oh I see what you meant now in the other PR :) if Tuples don't pollute the method arguments, I am ok with this, actually it simplifies synchronization issues between the two maps otherwise, I will update my PR to do the same.